bayes_opt/constraint.py

```diff
@@ -114,33 +162,53 @@ def predict(self, X):
                             if self._lb[0] != np.inf else np.array([1]))
             result = p_upper - p_lower
             return result.reshape(X_shape[:-1])
-        else:
-            result = np.ones(X.shape[0])
-            for j, gp in enumerate(self._model):
-                y_mean, y_std = gp.predict(X, return_std=True)
-                p_lower = (norm(loc=y_mean, scale=y_std).cdf(self._lb[j])
-                           if self._lb[j] != -np.inf else np.array([0]))
-                p_upper = (norm(loc=y_mean, scale=y_std).cdf(self._ub[j])
-                           if self._lb[j] != np.inf else np.array([1]))
-                result = result * (p_upper - p_lower)
-            return result.reshape(X_shape[:-1])
+
+        result = np.ones(X.shape[0])
+        for j, gp in enumerate(self._model):
+            y_mean, y_std = gp.predict(X, return_std=True)
+            p_lower = (norm(loc=y_mean, scale=y_std).cdf(self._lb[j])
+                        if self._lb[j] != -np.inf else np.array([0]))
+            p_upper = (norm(loc=y_mean, scale=y_std).cdf(self._ub[j])
+                        if self._lb[j] != np.inf else np.array([1]))
+            result = result * (p_upper - p_lower)
+        return result.reshape(X_shape[:-1])
 
     def approx(self, X):
         """
-        Returns the approximation of the constraint function using the internal
-        Gaussian Process Regressors.
+        Approximate the constraint function using the internal GPR model.
+
+        Parameters
+        ----------
+        X : np.ndarray of shape (n_samples, n_features)
+            Parameters for which to estimate the constraint function value.
+
+        Returns
+        -------
+        np.ndarray of shape (n_samples, n_constraints)
+            Constraint function value estimates.
         """
         X_shape = X.shape
         X = X.reshape((-1, self._model[0].n_features_in_))
         if len(self._model) == 1:
             return self._model[0].predict(X).reshape(X_shape[:-1])
-        else:
-            result = np.column_stack([gp.predict(X) for gp in self._model])
-            return result.reshape(X_shape[:-1] + (len(self._lb), ))
+
+        result = np.column_stack([gp.predict(X) for gp in self._model])
+        return result.reshape(X_shape[:-1] + (len(self._lb), ))
 
     def allowed(self, constraint_values):
-        """
-        Checks whether `constraint_values` are below the specified limits.
+        """Check whether `constraint_values` are below the specified limits.
```

Comment:
nit: very minor, I think "below" is not the right wording, since there is lb and ub. 

```suggestion
        """Check whether `constraint_values` respects the specified limits.
```

Does it make sense?